cppapi/server/multiattribute.cpp

```diff
@@ -936,8 +964,13 @@ void MultiAttribute::add_fwd_attribute(string &dev_name,DeviceClass *dev_class_p
 	vector<Attribute *>::iterator ite;
 	ite = attr_list.end() - 2;
 
-	attr_list.insert(ite,new FwdAttribute(prop_list,*new_attr,dev_name,index));
+	Attribute * new_fwd_attr = new FwdAttribute(prop_list,*new_attr,dev_name,index);
```

Comment:
Seems to me as a code duplication. Must be extracted into a dedicated method overloaded from `add_attr`, see below